actionpack/lib/action_controller/railtie.rb

```diff
@@ -67,5 +67,12 @@ class Railtie < Rails::Railtie #:nodoc:
         config.compile_methods! if config.respond_to?(:compile_methods!)
       end
     end
+
+    initializer "setup session store" do |app|
+      ActiveSupport.on_load(:action_controller) do
+        app_name = app.class.name ? app.railtie_name.sub(/_application$/, '') : ''
```

Comment:
For few tests I noticed that the app_name was coming out to be empty. I will revert this locally and post which were those tests.